examples/with-componentdidcatch/pages/_app.js

```diff
@@ -1,9 +1,48 @@
-import App from 'next/app'
+import App, { Container } from 'next/app'
+import React from 'react'
 
 export default class MyApp extends App {
+  static async getInitialProps ({ Component, router, ctx }) {
+    let pageProps = {}
+
+    if (Component.getInitialProps) {
+      pageProps = await Component.getInitialProps(ctx)
+    }
+
+    return { pageProps }
+  }
+
+  state = { error: null, errorInfo: null }
+
   componentDidCatch (error, errorInfo) {
-    console.log('CUSTOM ERROR HANDLING', error)
-    // This is needed to render errors correctly in development / production
-    super.componentDidCatch(error, errorInfo)
+    this.setState({ error, errorInfo })
+
+    // If you wish to render error with the usual error page, you can use :
+    // super.componentDidCatch(error, errorInfo)
```

Comment:
This is very important in development mode, where you want the debug component to render with all the information and source maps applied on the stack traces

Reply:
In production this will make sure _error.js is rendered.

Reply:
Which is the reason I only did a `console.log` in the example

Reply:
That makes sense. It's probably better to edit _error.js than handling the error here.

Reply:
Yep!

Reply:
I'm reverting _app.js back to its previous state 😙